ee/clickhouse/views/test/test_clickhouse_insights.py

```diff
@@ -497,3 +499,72 @@ def test_funnel_invalid_exclusions(self):
                 )
             else:
                 self.assertEqual(response.status_code, 200)
+
+    @patch("ee.clickhouse.views.insights.ClickhouseInsightsViewSet.calculate_funnel")
+    def test_that_multi_property_breakdown_is_not_breaking(self, mcf):
+
+        test_cases: List[Dict[str, Any]] = [
+            # single property
+            {"breakdown": "$browser", "funnel result": ["Chrome", "Safari"], "expected": ["Chrome", "Safari"]},
+            # single property client, multi property query result
+            {"breakdown": "$browser", "funnel result": [["Chrome"], ["Safari"]], "expected": ["Chrome", "Safari"]},
+            # multi property client, multi property query result
+            {
+                "breakdown": ["$browser"],
+                "funnel result": [["Chrome"], ["Safari"]],
+                "expected": [["Chrome"], ["Safari"]],
+            },
+        ]
+
+        for test_case in test_cases:
+
+            filter_with_breakdown = {
+                "insight": "FUNNELS",
+                "date_from": "-14d",
+                "actions": [],
+                "events": [
+                    {"id": "$pageview", "name": "$pageview", "type": "events", "order": 0},
+                    {"id": "$pageview", "type": "events", "order": 1, "name": "$pageview"},
+                ],
+                "display": "FunnelViz",
+                "interval": "day",
+                "properties": [],
+                "funnel_viz_type": "steps",
+                "exclusions": [],
+                "breakdown": test_case["breakdown"],
+                "breakdown_type": "event",
+                "funnel_from_step": 0,
+                "funnel_to_step": 1,
+            }
+
+            mcf.return_value = {"result": [[self.as_result(b), self.as_result(b)] for b in test_case["funnel result"]]}
+
+            response = self.client.post(f"/api/projects/{self.team.id}/insights/funnel", filter_with_breakdown)
+            self.assertEqual(200, response.status_code)
+
+            response_data = response.json()
+
+            result = response_data["result"]
+
+            # input events have chrome and safari so results is an array with two arrays as its contents
+            for i in range(0, 2):
+                for r in result[i]:
+                    self.assertIsInstance(r["name"], str)
+                    self.assertEqual(test_case["expected"][i], r["breakdown"])
+                    self.assertEqual(test_case["expected"][i], r["breakdown_value"])
+
+    @staticmethod
+    def as_result(b: Union[str, List[str]]) -> Dict[str, Any]:
+        return {
+            "action_id": "$pageview",
+            "name": "$pageview",
+            "custom_name": None,
+            "order": 0,
+            "people": ["a uuid"],
+            "count": 1,
+            "type": "events",
+            "average_conversion_time": None,
+            "median_conversion_time": None,
+            "breakdown": b,
```

Comment:
shouldn't this return `$browser` ?

Reply:
(more descriptive variable names here would be nice! This kinda made it hard for me to parse what values are being passed in)

Reply:
Ah, nevermind, looks like this is good: https://github.com/PostHog/posthog/pull/5495/files